src/backend/main.ts

```diff
@@ -791,106 +819,132 @@ ipcMain.handle(
   }
 )
 
-ipcMain.handle('getUserInfo', async () => LegendaryUser.getUserInfo())
+ipcMain.handle(
+  'getUserInfo',
+  async (): Promise<UserInfo | undefined> => LegendaryUser.getUserInfo()
+)
 
 // Checks if the user have logged in with Legendary already
-ipcMain.handle('isLoggedIn', async () => LegendaryUser.isLoggedIn())
+ipcMain.handle(
+  'isLoggedIn',
+  async (): Promise<boolean> => LegendaryUser.isLoggedIn()
+)
 
 ipcMain.handle('login', async (event, sid) => LegendaryUser.login(sid))
 ipcMain.handle('authGOG', async (event, code) => GOGUser.login(code))
 ipcMain.handle('logoutLegendary', async () => LegendaryUser.logout())
-ipcMain.handle('logoutGOG', async () => GOGUser.logout())
+ipcMain.on('logoutGOG', async () => GOGUser.logout())
```

Comment:
any reason this one changes from `handle` to `on`? I understand they are slightly different but I don't know why it's changed only for this logout and not the other

Reply:
is it because LegendaryUser.logout() is async and GOGUser.logout() is sync? I wonder if we should have both logout methods to be async for consistency

Reply:
Yup, that was the reason I've made that change. I was also thinking about leaving it as-is because of consistency, but decided against that since you can easily see which functions are async/sync in the frontend, so consistency isn't really needed